### PR TITLE
White soap bars are now createable via chemistry

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

Creates a new chemical recipe, 10 units of lye and 10 units of corn oil heated to 373K will create a standard white soap bar at the reaction site.

## Why It's Good For The Game

It's always important to keep labs squeaky clean. This will make soap more mass produce-able, but miners could already buy them for 200 points. This also requires cooperation from botany to actually get the corn oil.

## Testing

I made some soap with my new recipe, and make gibsoap to make sure I didn't accidentally break that in the process.
## Changelog
:cl:
add: 10 units each of lye and corn oil will now make a bar of soap when heated to 373K.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 1]
```diff
@@ -233,12 +233,23 @@
...
+	min_temp = T0C + 100
+	result_amount = 1
+	
+/datum/chemical_reaction/soapification/gib
+	name = "Gib Soapification"
+	id = "gib_soapification"
+	result = null
 	required_reagents = list("liquidgibs" = 10, "lye"  = 10) // requires two scooped gib tiles
 	min_temp = T0C + 100
 	result_amount = 1
```
Comment (Contrabang) [inline code/modules/reagents/chemistry/recipes/others.dm:246]:
```suggestion
	required_reagents = list("liquidgibs" = 10, "lye"  = 10) // requires two scooped gib tiles
```
result, min_temp, and result amount is carried over by the parent

**Comment by GDNgit** [comment]
Chemistry can already make soap though? 10 units of lye and 10 units of liquid gibs does the same thing, except makes a version of the soap that's *better* than this version.

**Comment by Sirryan2002** [comment]
Not a fan of this because it allows easy mass production of soap. Much like you can produce a shitton of cheese, all you need to do is to grow corn and find a chem dispenser and then you can make 10+ bars easily. I'd rather soap be a bit more rare as it negates needing to get the janitor to come.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -233,12 +233,20 @@
...
+	id = "gib_soapification"
+	required_reagents = list("liquidgibs" = 10, "lye"  = 10) // requires two scooped gib tiles
 
 /datum/chemical_reaction/soapification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	new /obj/item/soap/nanotrasen(location)
+
+/datum/chemical_reaction/soapification/gib/on_reaction(datum/reagents/holder, created_volume)
 	var/location = get_turf(holder.my_atom)
 	new /obj/item/soap/homemade(location)
```
Comment (hal9000PR) [inline code/modules/reagents/chemistry/recipes/others.dm:251]: you dont need 2 overrides, you could just have a var on the reaction thats like `soap_type` then new that.

**Pull request closed without merging**